robot/rover/CMakeLists.txt

```diff
@@ -0,0 +1,203 @@
+cmake_minimum_required(VERSION 3.5.0)
+set(CMAKE_SYSTEM_PROCESSOR arm)
+
+#execute a shell script to find the arduino directory. It uses the setup.sh file as every arduino version has that in there in the 1st level dir
+execute_process(
+COMMAND find / -name "arduino-linux-setup.sh" -print -quit
+OUTPUT_VARIABLE ARDUINO_SAMPLE_FILE
+OUTPUT_STRIP_TRAILING_WHITESPACE
+)
```

Comment:
pretty smart

Reply:
What happens if it doesnt find it? Could you print a message and exit? What happens if they have the ARDUINO IDE installed but not TeensyDuino?

Reply:
In retrospect, I think we should keep all the error handling for another PR because there's a lot of that, especially with making sure all libs are there etc.

Reply:
A suggestion could be to have a config file that the user just puts their path.

Reply:
My previous design had it so that a user entered his own path in the command line, like this : `cmake .. -DARDUINO="...path...."` would this idea be better? its just that its more work for the user I find

Reply:
I can also make it so that it only finds a Teensy folder, otherwise it lets the user knows, I just assumed everyone already had the Teensy one installed

Reply:
hmm if we set our ARDUINO_PATH in ~/.bashrc then the command would be

cmake .. -DARDUINO="$ARDUINO_PATH"

further we could just alias this to alias Build Robotics (BR)

```
alias br='cmake .. -DARDUINO="$ARDUINO_PATH"'
```

Reply:
then you just need to type `br` but idk if there's other suggestions @PeterGhimself @vashmata

Reply:
actually if you set the path in ~/.bashrc , CMake can read env variables right from there, so thats also an option, so it would still just be `cmake ..`

Reply:
Oh and for the libraries, I didn't add error handling code because there's most likely gonna be a scripted installed of those libraries in an issue, so that would handle any potential errors

Reply:
@MartensCedric Okay 2 options for this
- Find the correct Teensyduino folder, even if he has multiple Arduinos installed (like you did), and make sure to let the user know if he doesn't have the proper one installed
- Let the user enter the full path to Teensyduino in bashrc, this is also much easier to code, and more 'safe', as you can't really mess it up from the Cmake side as long as the user did their job right (it would let them know if the path is incorrect)

What's better?